packages/light-client/src/spec/utils.ts

```diff
@@ -96,6 +96,7 @@ export function upgradeLightClientHeader(
   const startUpgradeFromFork = Object.values(ForkName)[ForkSeq[headerFork] + 1];
 
   switch (startUpgradeFromFork) {
+    // biome-ignore lint/suspicious/useDefaultSwitchClauseLast: We want default to evaluate at first to throw error early
```

Comment:
I doubt this makes a difference where we put the `default` because it will have to evaluate all `case` either way

so just semantically it makes more sense to me to put the `default` last as it will always be called last if not `case` matches

Reply:
From the usage it seems author want to fail early without evaluation of the rest of the code. As all of the cases have fall-through behavior and if any case match the `default` will execute.

Reply:
[according to chatgpt](https://chatgpt.com/share/670cef74-1f48-8013-8360-18fc717f7c2f) the order of  `default` does not matter

> the default case will only be executed if no other case matches

Reply:
ChatGPT is wrong here. Check this [code](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBAhgLjAVwLYCMCmAnGBeGAJgG4YAoMiAdwEspgALGACjgEoYBvMmXmYOBEwwAjAnJ9J-cBBAAbTADo5IAObMARCI1tiPPgKFFx+qdMjylK9RsI69kgCaYAZnGRyoJs71AWFymqazm4eUPamUmQAvkA).

Reply:
yeah, in case of fall throug the order matters, kinda makes sense